content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
+
+However, we can see that the `return` keyword also has additional effects. As well as defining the output, it also tells us to "return" to the place in our code where the function was called.
+
+{{<note type="activity" title="exercise">}}
+Try using the visualisation diagram above to work out what happens when the computer executes the return statement.
+
+{{</note>}}
+
+### 🧩 Putting everything together
+
+So now we have some idea of how the computer executes our code, we can apply these ideas to a program for our current implementation of `convertToPercentage`:
+
+<iframe width="800" height="350" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=function%20convertToPercentage%28%29%20%7B%0A%20%20const%20num%20%3D%200.5%3B%0A%20%20const%20percentage%20%3D%20%60%24%7Bnum%20*%20100%7D%25%60%3B%0A%20%20return%20percentage%3B%0A%7D%0A%0Aconst%20output1%20%3D%20convertToPercentage%280.5%29%3B%0Aconst%20output2%20%3D%20convertToPercentage%280.3%29%3B&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=0&heapPrimitives=nevernest&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
+
+{{<note type="activity" title="exercise">}}
+
+Earlier we noted that when the computer executes the code above, we get the following output:
+
+```bash
+50%
+50%
+```
+
+Given the code above, try using the visualisation diagrams to "play computer" with this example. Try stepping through each line of code to explain why the program is giving this output
+
+{{</note>}}
+
+### Checking our code
+
+Given our code below:
+
+```js {linenos=table,hl_lines=[7] ,linenostart=1}
+function convertToPercentage() {
+  const num = 0.5;
+  const percentage = `${num * 100}%`;
+  return percentage;
+}
+
+const output1 = convertToPercentage(0.5);
+const output2 = convertToPercentage(0.23);
+```
+
+when the function `convertToPercentage` is called, the variable `num` gets recreated in the `convertToPercentage` frame and set to `0.5` every time we call the function. Instead of `num` being reset every time the function is called, we want to make use of the input that is passed to the function `convertToPercentage` every time it is called.
+
+### Parametrising a function
+
+To make this function reusable for any number, we need to handle inputs. We do this using a **parameter**.
+
+🔑 A parameter is a variable that enables us to reference the input to a function.
+
+We can add a parameter `num` to our function, as with the execution diagram below:
```

Comment:
We should make clear that we're _replacing_ the variable definition with a parameter with the same name. It could be easy to miss that up-lifting is happening. This would be a great place to add some explicit text along the lines of "A parameter is a variable whose value is defined by the caller - it's a variable just like the `num` variable was before, but now that it's a parameter we don't assign it a value in the function, the caller gives a value which will be assigned to it".

Reply:
Additionally I'd say we'd need to display the updated code in a visually consistent way, so I'd include  a code block of the likes:

```js {linenos=table,hl_lines=[7] ,linenostart=1}
function convertToPercentage(num) {   // now num is a parameter of convertToPercentage
  const percentage = `${num * 100}%`;
  return percentage;
}

const output1 = convertToPercentage(0.5);
const output2 = convertToPercentage(0.23);
```

I know the updated code is in the code visualizer a bit below, but I'm afraid the change can be easily missed by the different visualization, which may be distracting.